app/forms/search_form.rb

```diff
@@ -58,21 +64,28 @@ def statuses
     ].map { Option.new(_1, I18n.t("search.statuses.#{_1}")) }
   end
 
+  def application_types
+    self.class::APPLICATION_TYPES
+  end
+
   private
 
   def at_least_one_field_set
-    field_set = [:query, :submitted_from,
-                 :submitted_to, :updated_from,
-                 :updated_to, :status_with_assignment,
-                 :caseworker_id].any? do |field|
+    fields = [:query, :submitted_from,
+              :submitted_to, :updated_from,
+              :updated_to, :status_with_assignment,
+              :caseworker_id]
+
+    fields.append(:application_type) if form_context == 'analytics' && FeatureFlags.nsm_insights.enabled?
```

Comment:
i agreed with you and then realised that this depends on what logic is desired for the page
application_type is always present in a service context, and it's one of the fields that at_least_one_field_set checks for, then the form will be considered valid if you don't enter details - this is fine if we think it's ok that not entering anything should return all crm4 applications - but I had made an assumption that users would need to apply a filter

In the same vein, if no filter in service context returns all crm4s, should no filter in analytics context return all submissions?

Reply:
Ohh, yeah my bad, I misread my own code! Yes, we do need users to enter something on the Prior Authority search page, there is an explicit requirement that if they leave everything blank we show them an error message. Whereas in the analytics context, if they leave everything blank _except_ the application type, then it might be we don't want to show them an error. So depending on context, application type potentially should be one of the "at least one of these fields must be set".

If that's the route we go down, then I don't think the SearchForm should _know_ about what the different contexts are - it shouldn't know that the requirements for the analytics screen are different to the PA search screen. Instead, I think in those contexts should explicitly tell the SearchForm what their requirements are. E.g. instead of the controller telling SearchForm "this is the analytics context, I expect you to know the ramifications of that", it should tell it "this is a context where application_type is not one of the fields that at least one of needs to be set".

So maybe it's an "explicit_application_type" boolean rather than a "form_context" string?